src/iceberg/transaction.h

```diff
@@ -39,6 +39,76 @@ class ICEBERG_EXPORT Transaction {
   /// \return this transaction's table
   virtual const std::shared_ptr<Table>& table() const = 0;
 
+  /// \brief Create a new schema addition operation
+  ///
+  /// \return a new AddSchema
+  virtual std::shared_ptr<AddSchema> AddSchema() = 0;
```

Comment:
```suggestion
  virtual std::unique_ptr<AddSchema> AddSchema() = 0;
```

I'd suggest returning std::unique_ptr for all actions here and below.